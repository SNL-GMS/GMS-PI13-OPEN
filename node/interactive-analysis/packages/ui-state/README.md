## UI State

## Build Scripts
  * `precommit`: executed when files are staged for commit (git add) for prettify; this currently is not enabled,
  * `clean:node_modules`: removes the node_modules directory
  * `clean:build-scripts`: removes the (copied over) common build scripts directory
  * `clean:dist`: removes all of the build directories (build/, dist/, bundle-analyzer/)
  * `clean:coverage`: removes the coverage directory 
  * `clean:docs`: removes the documents directory
  * `clean`: cleans and removes all
  * `build:build-scripts`: copies over the common build scripts
  * `build:tslint`: runs tslint checks on the package
  * `build:tsc`: compiles the typescript project and generates the type definitions
  * `build:tsc-defs`: generates the type definitions
  * `build:dev`: runs the development build of the package, includes source maps
  * `build:prod`: runs the production build of the package, does not include source maps (preforms minification)
  * `build`: runs the production build of the package
  * `start:dev`: starts development mode `localhost:8080/`
  * `start:prod`: starts in production mode `localhost:8080/`
  * `dev`: starts in development mode `localhost:8080/`
  * `start`: starts in production mode `localhost:8080/`
  * `watch:dev`: runs the development build of the package and watches for changes to recompile
  * `watch:prod`: runs the production build of the package and watches for changes to recompile
  * `watch`: runs the production build and watches for changes to recompile
  * `docs`: generates the package source documentation
  * `sonar`: runs sonar lint checks
  * `test`: runs the package jest tests
  * `version`: returns the version of the package

## Documentation

To generate HTML documentation files:
```
$ yarn docs
```

## Building

```bash
[.../common-util] $ yarn build
```

## Generating Binaries

```bash
[.../common-util] $ yarn build
```