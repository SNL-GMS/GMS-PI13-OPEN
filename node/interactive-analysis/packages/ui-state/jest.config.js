const { resolve } = require('path');
const root = resolve(__dirname, '.');

module.exports = {
  rootDir: root,
  verbose: false,
  bail: true,
  silent: true,
  cacheDirectory: '<rootDir>/.cache/jest',
  name: 'ui-state',
  testURL: 'http://localhost/',
  globals: {
    'ts-jest': {
      diagnostics: false
    }
  },
  setupFiles: ['<rootDir>/jest.setup.ts'],
  snapshotSerializers: ['enzyme-to-json/serializer'],
  transform: {
    '^.+\\.jsx?$': 'babel-jest',
    '^.+\\.tsx?$': 'ts-jest'
  },
  testRegex: '/__tests__/.*\\.(test|spec)\\.(ts|tsx)$',
  moduleFileExtensions: ['ts', 'tsx', 'js', 'json'],
  modulePaths: ['./node_modules'],
  moduleDirectories: ['./node_modules'],
  modulePathIgnorePatterns: [
    '<rootDir>/vscode',
    '<rootDir>/coverage',
    '<rootDir>/dist/',
    '<rootDir>/node_modules/',
    '<rootDir>/resources/'
  ],
  moduleNameMapper: {
    '@gms/([^/]+)$': '<rootDir>/../$1/src/ts/$1',
    '@gms/([^/]+)(/lib/)(.*)$': '<rootDir>/../$1/src/ts/$3',
    '.*\\.(css|less|styl|scss|sass)$': '<rootDir>/__mocks__/style-mock.ts',
    '.*\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/file-mock.ts'
  },
  testEnvironment: 'jest-environment-jsdom-fourteen',
  collectCoverage: true,
  coverageReporters: ['lcov', 'html']
};
