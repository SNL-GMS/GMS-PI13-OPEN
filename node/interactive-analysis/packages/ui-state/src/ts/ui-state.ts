export * from './state';
export * from './state/keypress';
