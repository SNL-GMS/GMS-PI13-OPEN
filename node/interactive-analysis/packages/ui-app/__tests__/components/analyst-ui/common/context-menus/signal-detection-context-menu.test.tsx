import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import React from 'react';
import {
  SignalDetectionContextMenu,
  SignalDetectionContextMenuProps
} from '../../../../../src/ts/components/analyst-ui/common/context-menus/signal-detection-context-menu';
import { signalDetectionsData } from '../../../../__data__/signal-detections-data';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

function flushPromises(): any {
  return new Promise(setImmediate);
}

// set up window alert and open so we don't see errors
window.alert = jest.fn();
window.open = jest.fn();
/**
 * Tests the signal detection context menu component
 */
describe('signal-detection-context-menu', () => {
  // tslint:disable-next-line: one-variable-per-declaration
  const props: SignalDetectionContextMenuProps = {
    signalDetections: signalDetectionsData,
    selectedSds: [signalDetectionsData[0]],
    sdIdsToShowFk: [],
    currentOpenEvent: null,
    changeAssociation: jest.fn(),
    associateToNewEvent: jest.fn(),
    rejectDetections: jest.fn(),
    updateDetections: jest.fn(),
    measurementMode: {
      mode: AnalystWorkspaceTypes.WaveformDisplayMode.DEFAULT,
      entries: Immutable.Map<string, boolean>().set(signalDetectionsData[0].id, true)
    },
    setSelectedSdIds: jest.fn(),
    setSdIdsToShowFk: jest.fn(),
    setMeasurementModeEntries: jest.fn()
  };
  const wrapper: any = Enzyme.mount(
    <SignalDetectionContextMenu
      signalDetections={props.signalDetections}
      selectedSds={props.selectedSds}
      sdIdsToShowFk={props.sdIdsToShowFk}
      currentOpenEvent={props.currentOpenEvent}
      changeAssociation={props.changeAssociation}
      associateToNewEvent={props.associateToNewEvent}
      rejectDetections={props.rejectDetections}
      updateDetections={props.updateDetections}
      measurementMode={props.measurementMode}
      // tslint:disable-next-line: no-unbound-method
      setSelectedSdIds={props.setSelectedSdIds}
      // tslint:disable-next-line: no-unbound-method
      setSdIdsToShowFk={props.setSdIdsToShowFk}
      // tslint:disable-next-line: no-unbound-method
      setMeasurementModeEntries={props.setMeasurementModeEntries}
    />
  );

  it('should have a consistent snapshot on mount', () => {
    expect(wrapper.render()).toMatchSnapshot();
    flushPromises();
  });

  it('should have basic props built on render', () => {
    const buildProps = wrapper.props() as SignalDetectionContextMenuProps;
    expect(buildProps).toMatchSnapshot();
  });

  it('should have a function for checking if we can generate fk', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'canGenerateFk');
    expect(instance.canGenerateFk(signalDetectionsData[0])).toMatchSnapshot();
    flushPromises();
    expect(spy).toHaveBeenCalledWith(signalDetectionsData[0]);
  });

  it('should have a function for setSdIdsToShowFk', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'setSdIdsToShowFk');
    instance.setSdIdsToShowFk();
    flushPromises();
    expect(spy).toHaveBeenCalled();
    // if we call setSdIdsToShowFk it should call set selected sd ids
    // tslint:disable-next-line: no-unbound-method
    expect(props.setSelectedSdIds).toHaveBeenCalledWith([signalDetectionsData[0].id]);
  });
});
