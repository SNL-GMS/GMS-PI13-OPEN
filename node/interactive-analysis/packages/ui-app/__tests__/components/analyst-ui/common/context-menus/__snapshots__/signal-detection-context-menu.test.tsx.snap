// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signal-detection-context-menu should have a consistent snapshot on mount 1`] = `
<ul
  class="bp3-menu"
>
  <li
    class="bp3-submenu"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <a
          class="bp3-menu-item"
          data-cy="set-phase"
          label="Ctrl+s"
          tabindex="0"
        >
          <div
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Set Phase...
          </div>
          <span
            class="bp3-menu-item-label"
          >
            Ctrl+s
          </span>
          <span
            class="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    class="bp3-submenu"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <a
          class="bp3-menu-item"
          data-cy="association-menu"
          tabindex="0"
        >
          <div
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Event Association
          </div>
          <span
            class="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    class=""
  >
    <a
      class="bp3-menu-item bp3-popover-dismiss"
      data-cy="show-fk"
    >
      <div
        class="bp3-text-overflow-ellipsis bp3-fill"
      >
        Show FK
      </div>
    </a>
  </li>
  <li
    class=""
  >
    <a
      class="bp3-menu-item bp3-popover-dismiss"
      data-cy="reject-sd"
    >
      <div
        class="bp3-text-overflow-ellipsis bp3-fill"
      >
        Reject
      </div>
    </a>
  </li>
  <li
    class="bp3-submenu"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <a
          class="bp3-menu-item"
          data-cy="measure"
          tabindex="0"
        >
          <div
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Measure
          </div>
          <span
            class="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
</ul>
`;

exports[`signal-detection-context-menu should have a function for checking if we can generate fk 1`] = `true`;

exports[`signal-detection-context-menu should have basic props built on render 1`] = `
Object {
  "associateToNewEvent": [MockFunction],
  "changeAssociation": [MockFunction],
  "currentOpenEvent": null,
  "measurementMode": Object {
    "entries": Immutable.Map {
      "acbd4ff2-f6e1-4a50-bc43-8ccc4f274b05": true,
    },
    "mode": "Default",
  },
  "rejectDetections": [MockFunction],
  "sdIdsToShowFk": Array [],
  "selectedSds": Array [
    Object {
      "conflictingHypotheses": Array [],
      "currentHypothesis": Object {
        "featureMeasurements": Array [
          Object {
            "featureMeasurementType": "ARRIVAL_TIME",
            "id": "dbdd8420-a448-44a0-9fea-d39fe55b0136",
            "measurementValue": Object {
              "standardDeviation": 0.685,
              "value": 1274393237.85,
            },
          },
          Object {
            "featureMeasurementType": "RECEIVER_TO_SOURCE_AZIMUTH",
            "id": "3f9b417b-bc12-44ad-855d-132355708ec5",
            "measurementValue": Object {
              "measurementValue": 0,
              "referenceTime": 0,
            },
          },
          Object {
            "featureMeasurementType": "SLOWNESS",
            "id": "3f9b417b-bc12-44ad-855d-132355708ec5",
            "measurementValue": Object {
              "measurementValue": 0,
              "referenceTime": 0,
            },
          },
          Object {
            "featureMeasurementType": "PHASE",
            "id": "3f9b417b-bc12-44ad-855d-132355708ec5",
            "measurementValue": Object {
              "confidence": 0,
              "phase": "P",
            },
          },
        ],
        "id": "acbd4ff2-f6e1-4a50-bc43-8ccc4f274b05",
        "rejected": false,
      },
      "hasConflict": false,
      "id": "acbd4ff2-f6e1-4a50-bc43-8ccc4f274b05",
      "modified": false,
      "monitoringOrganization": "TEST",
      "requiresReview": Object {
        "amplitudeMeasurement": false,
      },
      "reviewed": Object {
        "amplitudeMeasurement": false,
      },
      "signalDetectionHypothesisHistory": Array [
        Object {
          "arrivalTimeSecs": 1274393334.1,
          "arrivalTimeUncertainty": 0.685,
          "id": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
          "phase": "tx",
          "rejected": false,
        },
        Object {
          "arrivalTimeSecs": 1274393334.1,
          "arrivalTimeUncertainty": 0.685,
          "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
          "phase": "P",
          "rejected": false,
        },
      ],
      "stationName": "USRK",
    },
  ],
  "setMeasurementModeEntries": [MockFunction],
  "setSdIdsToShowFk": [MockFunction],
  "setSelectedSdIds": [MockFunction],
  "signalDetections": Array [
    Object {
      "conflictingHypotheses": Array [],
      "currentHypothesis": Object {
        "featureMeasurements": Array [
          Object {
            "featureMeasurementType": "ARRIVAL_TIME",
            "id": "dbdd8420-a448-44a0-9fea-d39fe55b0136",
            "measurementValue": Object {
              "standardDeviation": 0.685,
              "value": 1274393237.85,
            },
          },
          Object {
            "featureMeasurementType": "RECEIVER_TO_SOURCE_AZIMUTH",
            "id": "3f9b417b-bc12-44ad-855d-132355708ec5",
            "measurementValue": Object {
              "measurementValue": 0,
              "referenceTime": 0,
            },
          },
          Object {
            "featureMeasurementType": "SLOWNESS",
            "id": "3f9b417b-bc12-44ad-855d-132355708ec5",
            "measurementValue": Object {
              "measurementValue": 0,
              "referenceTime": 0,
            },
          },
          Object {
            "featureMeasurementType": "PHASE",
            "id": "3f9b417b-bc12-44ad-855d-132355708ec5",
            "measurementValue": Object {
              "confidence": 0,
              "phase": "P",
            },
          },
        ],
        "id": "acbd4ff2-f6e1-4a50-bc43-8ccc4f274b05",
        "rejected": false,
      },
      "hasConflict": false,
      "id": "acbd4ff2-f6e1-4a50-bc43-8ccc4f274b05",
      "modified": false,
      "monitoringOrganization": "TEST",
      "requiresReview": Object {
        "amplitudeMeasurement": false,
      },
      "reviewed": Object {
        "amplitudeMeasurement": false,
      },
      "signalDetectionHypothesisHistory": Array [
        Object {
          "arrivalTimeSecs": 1274393334.1,
          "arrivalTimeUncertainty": 0.685,
          "id": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
          "phase": "tx",
          "rejected": false,
        },
        Object {
          "arrivalTimeSecs": 1274393334.1,
          "arrivalTimeUncertainty": 0.685,
          "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
          "phase": "P",
          "rejected": false,
        },
      ],
      "stationName": "USRK",
    },
    Object {
      "conflictingHypotheses": Array [],
      "currentHypothesis": Object {
        "featureMeasurements": Array [
          Object {
            "featureMeasurementType": "ARRIVAL_TIME",
            "id": "e8d95a93-382d-4d84-85ce-983e87bdcb0c",
            "measurementValue": Object {
              "standardDeviation": 0.685,
              "value": 1274393334.1,
            },
          },
          Object {
            "featureMeasurementType": "RECEIVER_TO_SOURCE_AZIMUTH",
            "id": "8cdefd4e-803f-48f1-9e51-4229066f9991",
            "measurementValue": Object {
              "measurementValue": 0,
              "referenceTime": 0,
            },
          },
          Object {
            "featureMeasurementType": "SLOWNESS",
            "id": "8cdefd4e-803f-48f1-9e51-4229066f9991",
            "measurementValue": Object {
              "measurementValue": 0,
              "referenceTime": 0,
            },
          },
          Object {
            "featureMeasurementType": "PHASE",
            "id": "8cdefd4e-803f-48f1-9e51-4229066f9991",
            "measurementValue": Object {
              "confidence": 0,
              "phase": "P",
            },
          },
        ],
        "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
        "rejected": false,
      },
      "hasConflict": false,
      "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
      "modified": false,
      "monitoringOrganization": "TEST",
      "requiresReview": Object {
        "amplitudeMeasurement": false,
      },
      "reviewed": Object {
        "amplitudeMeasurement": false,
      },
      "signalDetectionHypothesisHistory": Array [
        Object {
          "arrivalTimeSecs": 1274393334.1,
          "arrivalTimeUncertainty": 0.685,
          "id": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
          "phase": "tx",
          "rejected": false,
        },
        Object {
          "arrivalTimeSecs": 1274393334.1,
          "arrivalTimeUncertainty": 0.685,
          "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
          "phase": "P",
          "rejected": false,
        },
      ],
      "stationName": "SONM",
    },
  ],
  "updateDetections": [MockFunction],
}
`;
